source/extensions/filters/http/transformation/body_header_transformer.cc

```diff
@@ -13,7 +13,8 @@ namespace Extensions {
 namespace HttpFilters {
 namespace Transformation {
 
-BodyHeaderTransformer::BodyHeaderTransformer(bool add_request_metadata):add_request_metadata_(add_request_metadata){}
+BodyHeaderTransformer::BodyHeaderTransformer(bool add_request_metadata, google::protobuf::BoolValue log_request_response_info)
```

Comment:
A drive-by comment: using `google::protobuf::BoolValue` feels a bit weird here, why not use `bool` instead? This also complicates code downstream, as now a check must be made if log_request_response_info contains a value before checking the value itself. 

Another nit with this is that `log_request_response_info` is being passed by value, which can be expensive with protobufs, as it tries to hold on to any allocated memory, even after objects are cleared/destroyed.

Reply:
Kevin suggested using `BoolValue` here https://github.com/solo-io/envoy-gloo/pull/240#discussion_r1207081565
Doing so allows the transformation-level setting to always take precedence over the listener-level setting, if defined

Reply:
Ty for explanation, that makes sense. A comment there might be useful to clear this up. Fwiw, `absl::optional<bool>` would work here, but at this point it would mostly be about not allocating lots of small protobuf objects.

Reply:
This just merged -- I'm a little surprised because I thought the work-in-progress tag would prevent that -- I'm going to put up a follow-up PR that clarifies this point and passes around `log_request_response_info` by reference after I handle the necessary envoy-gloo-ee changes